Last few betas do not include osx-amd64 version

As in the title.

For some reason osx-amd64 build for rclone beta is failing for the last few versions (since beta.8729.fe880c0fa):

```
~$ /usr/local/bin/rclone-beta selfupdate --beta -vv
2025/05/12 05:48:17 DEBUG : rclone: Version "v1.70.0-beta.8729.fe880c0fa" starting with parameters ["/usr/local/bin/rclone-beta" "selfupdate" "--beta" "-vv"]
2025/05/12 05:48:18 CRITICAL: Error: failed to update rclone: failed with 404 Not Found downloading https://beta.rclone.org/v1.70.0-beta.8734.68a65e878/rclone-v1.70.0-beta.8734.68a65e878-osx-amd64.zip
```

Right, I see it has been failing on build of master for some days..

From the logs:

```
go run bin/cross-compile.go -release beta-latest -git-log /tmp/git-log.txt -include "^darwin/amd64" -cgo -tags "cmount"  v1.70.0-beta.8734.68a65e878
2025/05/09 07:43:33 Compiling darwin/amd64 into rclone-v1.70.0-beta.8734.68a65e878-osx-amd64
2025/05/09 07:44:10 ----------------------------
2025/05/09 07:44:10 Failed to run [go build --ldflags -s -X github.com/rclone/rclone/fs.Version=v1.70.0-beta.8734.68a65e878 -trimpath -o rclone-v1.70.0-beta.8734.68a65e878-osx-amd64/rclone -tags cmount ..]: exit status 1
2025/05/09 07:44:10 Command output was:
go: downloading github.com/hanwen/go-fuse/v2 v2.7.2
# github.com/rclone/rclone
link: duplicated definition of symbol dlopen, from github.com/ebitengine/purego and github.com/ebitengine/purego
2025/05/09 07:44:10 ----------------------------
[20](https://github.com/rclone/rclone/actions/runs/14923834408/job/41924134439#step:15:21)25/05/09 07:44:10 Error compiling darwin/amd64: exit status 1
2025/05/09 07:44:10 Compiled 1 arches in 36.794499s
2025/05/09 07:44:10 1 compile failures:
  darwin/amd64
exit status 1
make: *** [ci_beta] Error 1
```

This matches this issue https://github.com/ebitengine/purego/issues/313 on one of our dependencies, and related to the latest Go version - 1.24.3. It has been fixed with https://github.com/ebitengine/purego/pull/312, released as [v0.8.3](https://github.com/ebitengine/purego/releases/tag/v0.8.3), so we should update to this.

--- a/go.mod
+++ b/go.mod
@@ -138,7 +138,7 @@ require (
 	github.com/cronokirby/saferith v0.33.0 // indirect
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc // indirect
 	github.com/dustin/go-humanize v1.0.1 // indirect
-	github.com/ebitengine/purego v0.8.2 // indirect
+	github.com/ebitengine/purego v0.8.3 // indirect
 	github.com/emersion/go-message v0.18.0 // indirect
 	github.com/emersion/go-textwrapper v0.0.0-20200911093747-65d896831594 // indirect
 	github.com/emersion/go-vcard v0.0.0-20230815062825-8fda7d206ec9 // indirect

